src/Common.sol

```diff
@@ -691,4 +691,19 @@ abstract contract Common is AccessControl, Pausable {
     function getMaxFeeX64(FeeType feeType) public view returns (uint64) {
         return _maxFeeX64[feeType];
     }
+
+    /// @dev some tokens require allowance == 0 to approve new amount
+    /// but some tokens does not aloow approve ammount = 0
+    /// we try to set allowence = 0 before approve new amount. if it revert means that
+    /// the token not allow to approve 0, which means the following line code will work properly
+    function _safeApprove(IERC20 token, address _spender, uint256 _value) internal {
+        bytes memory approveData = abi.encodeWithSelector(token.approve.selector, spender, 0);
```

Comment:
prove me if I'm wrong but I feel like approve to 0 is not our concern here. Since we always spend all allowance in the execution, token approval is considered reset at the end of the transaction. The only concern to me is the approve function interface, which you did address below

Reply:
Approve to 0 is needed because we might not use all `total0` (L: 490) to add to position.
We can reset at the end of transaction but we also can reset and approve new amount.

Reply:
okay